### PR TITLE
fix(portal): Fixing a few edge cases for identity email

### PR DESCRIPTION
**Pull request by bmanifold**

### CODE REVIEW

File: elixir/apps/domain/lib/domain/auth.ex [hunk 1]
```diff
@@ -914,4 +919,51 @@ defmodule Domain.Auth do
...
+      |> to_string
+      |> String.trim()
+
+    identifier =
+      params["provider_identifier"]
+      |> to_string()
+      |> String.trim()
+
+    cond do
+      Domain.Auth.valid_email?(email) ->
```
Comment (jamilbk) [inline elixir/apps/domain/lib/domain/auth.ex]:
```suggestion
      valid_email?(email) ->
```

File: elixir/apps/domain/lib/domain/auth.ex [hunk 2]
```diff
@@ -914,4 +919,51 @@ defmodule Domain.Auth do
...
+        params
+
+      Domain.Auth.valid_email?(identifier) ->
```
Comment (jamilbk) [inline elixir/apps/domain/lib/domain/auth.ex]:
```suggestion
      valid_email?(identifier) ->
```

File: elixir/apps/domain/lib/domain/auth.ex [hunk 3]
```diff
@@ -914,4 +919,51 @@ defmodule Domain.Auth do
...
+    identifier =
+      params["provider_identifier"]
+      |> to_string()
+      |> String.trim()
+
+    cond do
+      identifier != "" ->
+        params
+
+      Domain.Auth.valid_email?(email) ->
```
Comment (jamilbk) [inline elixir/apps/domain/lib/domain/auth.ex]:
```suggestion
      valid_email?(email) ->
```

File: elixir/apps/domain/lib/domain/auth/identity/changeset.ex [hunk 4]
```diff
@@ -67,7 +67,7 @@ defmodule Domain.Auth.Identity.Changeset do
       name: :auth_identities_account_id_provider_id_provider_identifier_idx
     )
     |> unique_constraint(:email,
-      name: :auth_identities_account_id_provider_id_email_idx
+      name: :auth_identities_acct_id_provider_id_email_prov_ident_unique_idx
```
Comment (jamilbk) [inline elixir/apps/domain/lib/domain/auth/identity/changeset.ex:70]: Whoops, good catch!

File: elixir/apps/domain/priv/repo/migrations/20241216183520_migrate_email_data_again.exs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      UPDATE auth_identities AS ai
+      SET email = COALESCE(
+        CASE
+          WHEN p.adapter = 'email' OR p.adapter = 'userpass' THEN ai.provider_identifier
+          ELSE COALESCE(
+            provider_state #>> '{claims,email}',
+            provider_state #>> '{userinfo,email}'
+          )
+        END
+      )
```
Comment (jamilbk) [inline elixir/apps/domain/priv/repo/migrations/20241216183520_migrate_email_data_again.exs]: Is the outer COALESCE needed?

```suggestion
      SET email = 
        CASE
          WHEN p.adapter = 'email' OR p.adapter = 'userpass' THEN ai.provider_identifier
          ELSE COALESCE(
            provider_state #>> '{claims,email}',
            provider_state #>> '{userinfo,email}'
          )
        END
```

**Review by jamilbk — APPROVED** [review]
Just some minor cleanup suggestions.

File: elixir/apps/domain/priv/repo/migrations/20241216183520_migrate_email_data_again.exs (hunk 5 above)
Comment (bmanifold) [reply elixir/apps/domain/priv/repo/migrations/20241216183520_migrate_email_data_again.exs]: Ah, good catch, I don't think it is.  I just copy/pasted that from the first one, but I'm not sure how that outer COALESCE got in the first one to begin with.  🤷‍♂️ 

Should be updated now.

File: elixir/apps/domain/lib/domain/auth.ex (hunk 2 above)
Comment (bmanifold) [reply elixir/apps/domain/lib/domain/auth.ex]: Ah, good catch!  Should be updated.

**Review by jamilbk — APPROVED** [review]

**Pull request merged**